oneflow/core/functional/impl/math_functor.cpp

```diff
@@ -77,25 +77,31 @@ class ScalarAddFunctor {
   Maybe<Tensor> operator()(const std::shared_ptr<one::Tensor>& x, const Scalar& scalar,
                            bool inplace) const {
     MutableAttrMap attrs;
+    TensorProcessor tensor_processor;
+    Symbol<DType> scalar_dtype;
     if (scalar.IsFloatingPoint()) {
       JUST(attrs.SetAttr<double>("float_operand", JUST(scalar.As<double>())));
       JUST(attrs.SetAttr<bool>("has_float_operand", true));
       JUST(attrs.SetAttr<bool>("has_int_operand", false));
+      scalar_dtype = DType::Double();
     } else if (scalar.IsIntegral()) {
       JUST(attrs.SetAttr<int64_t>("int_operand", JUST(scalar.As<int64_t>())));
       JUST(attrs.SetAttr<bool>("has_float_operand", false));
       JUST(attrs.SetAttr<bool>("has_int_operand", true));
+      scalar_dtype = DType::Int64();
     } else {
       UNIMPLEMENTED_THEN_RETURN() << "The scalar in ScalarAdd shoule be float or int.";
     }
```

Comment:
```suggestion
      Symbol<DType> lowest_dtype;
      if (scalar.IsFloatingPoint()) {
        JUST(attrs.SetAttr<double>("float_operand", JUST(scalar.As<double>())));
        JUST(attrs.SetAttr<bool>("has_float_operand", true));
        JUST(attrs.SetAttr<bool>("has_int_operand", false));
      if (DType::priority_order[*x->dtype()] < DType::priority_order[DType::Float16()]) {
        lowest_dtype = DType::Float();
      } else {
        lowest_dtype = x->dtype();
      }
    } else if (scalar.IsIntegral()) {
      JUST(attrs.SetAttr<int64_t>("int_operand", JUST(scalar.As<int64_t>())));
      JUST(attrs.SetAttr<bool>("has_float_operand", false));
      JUST(attrs.SetAttr<bool>("has_int_operand", true));
      lowest_dtype = x->dtype();
    } else {
      UNIMPLEMENTED_THEN_RETURN() << "The scalar in ScalarAdd shoule be float or int.";
    }
```

如果这样写呢？和 torch 会有行为不一致的情况吗？

Reply:
按这种方式处理，并且可以不用给tensorprocessor增加额外接口